test/functional/ES512TokenTest.php

```diff
@@ -0,0 +1,280 @@
+<?php
+declare (strict_types=1);
+
+namespace Lcobucci\JWT\FunctionalTests;
+
+use InvalidArgumentException;
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Keys;
+use Lcobucci\JWT\Signer\Ecdsa\Sha256;
+use Lcobucci\JWT\Signer\Ecdsa\Sha512;
+use Lcobucci\JWT\Signer\Key;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Validation\Constraint\SignedWith;
+use Lcobucci\JWT\Validation\InvalidToken;
+use PHPUnit\Framework\TestCase;
+
+class ES512TokenTest extends TestCase
+{
+    use Keys;
+
+    /**
+     * @var Configuration
+     */
+    private $config;
+
+    /**
+     * @before
+     */
+    public function createConfiguration(): void
+    {
+        $this->config = Configuration::forAsymmetricSigner(
+            Sha512::create(),
+            static::$ecdsaKeys['private_ec512'],
+            static::$ecdsaKeys['public_ec512']
+        );
+    }
+
+    /**
+     * @test
+     *
+     * @covers \Lcobucci\JWT\Configuration
```

Comment:
Hmm, I'd leave this at class level, and only put coverage on the ecdsa classes

Reply:
We should do this for all functional tests indeed, it would simplify everything. Sounds like a different PR, though :+1: